price_parser/parser.py

```diff
@@ -283,3 +292,31 @@ def parse_number(num: str) -> Optional[Decimal]:
         return Decimal(num)
     except InvalidOperation:
         return None
+
+
+def date_format(price):
+    for fmt in ['%d.%m.%Y', '%B, %Y', '%b, %Y', '%Y-%m-%d']:
+        try:
+            date = datetime.strptime(price, fmt)
+            if isinstance(date, datetime):
+                return date
+        except (ValueError, TypeError):
+            continue
+
+    return None
+
+
+def strip_date(text):
+    # normalize whitspace
+    text = re.sub(r'\s+', ' ', text)
+    all_date_regexp = [
+        r'\d{1,4}-\d{1,2}-\d{2,4}',
+        r' \S{3,8},\s\d{4}',
+    ]
+    text_processed = text
+    for regexp in all_date_regexp:
+        for match in re.finditer(regexp, text):
+            if match and date_format(match.group(0).strip()):
```

Comment:
I think we can assume `bool(match)` evaluated to `True` here.

```suggestion
            if date_format(match.group(0).strip()):
```

Reply:
Add more tests in https://github.com/scrapinghub/price-parser/pull/19/commits/20ac018ad2b57a9316da239d38e1feda69661735#diff-091825c38112ff1dafc081caf2916d27R2033